### PR TITLE
ci: enable AArch64 target in MSAN jobs

### PR DESCRIPTION
**Pull request by fanquake**
Make it possible to run the MSAN jobs on aarch64, as it was previously.

### CODE REVIEW

File: ci/test/01_base_install.sh [hunk 1]
```diff
@@ -42,11 +42,11 @@ if [ -n "$PIP_PACKAGES" ]; then
 fi
 
 if [[ ${USE_MEMORY_SANITIZER} == "true" ]]; then
-  git clone --depth=1 https://github.com/llvm/llvm-project -b llvmorg-16.0.4 "${BASE_SCRATCH_DIR}"/msan/llvm-project
+  git clone --depth=1 https://github.com/llvm/llvm-project -b llvmorg-16.0.5 "${BASE_SCRATCH_DIR}"/msan/llvm-project
 
   cmake -G Ninja -B "${BASE_SCRATCH_DIR}"/msan/clang_build/ -DLLVM_ENABLE_PROJECTS="clang" \
                                                             -DCMAKE_BUILD_TYPE=Release \
-                                                            -DLLVM_TARGETS_TO_BUILD=X86 \
+                                                            -DLLVM_TARGETS_TO_BUILD="X86;AArch64" \
```
[the remark this answers is not in the document]
Comment (hebasto) [reply ci/test/01_base_install.sh:49]: > Would be nice to just detect and build for the current CPU, if that is possible?

What about not providing the value at all, and using the default one?

[the remark this answers is not in the document]
Comment (fanquake) [reply ci/test/01_base_install.sh]: > if that is possible?

Done with `Native`.

**Review by dergoegge — APPROVED** [review]
utACK 2ebeb421dd21a69344a32c681ccbfcf5e5c6dab9

**Review by Dezzj — APPROVED** [review]
Approved

**Review by hebasto — COMMENTED** [review]
Post-merge ACK 2ebeb421dd21a69344a32c681ccbfcf5e5c6dab9. Tested on Ubuntu 23.04, `aarch64`.

FWIW, the [default](https://github.com/bitcoin/bitcoin/pull/27824#discussion_r1217967152) value, i.e., not setting `LLVM_TARGETS_TO_BUILD` altogether, works as well.

**Pull request merged**